frappe/database/query.py

```diff
@@ -506,12 +517,32 @@ def get_fieldnames_from_child_table(self, doctype, fields):
 				field = f"`tab{self.linked_doctype}`.`{linked_fieldname}`"
 				if alias:
 					field = f"{field} as {alias}"
-				fields[idx] = field
+				replacements[idx] = field
+
+		return replace_fields(replacements, fields)
+
+	def sanitize_fields(self, fields: str | list | tuple):
+		replacements = {}
+		if isinstance(fields, (list, tuple)):
+			fields = list(fields)
+			for idx, field in enumerate(fields):
+				if isinstance(field, str):
+					field = MARIADB_SPECIFIC_COMMENT.sub(
+						"", sqlparse.format(field, strip_comments=True, keyword_case="lower")
+					)
+					replacements[idx] = field
```

Comment:
I still don't see point of all this. 

You can just construct the new return list as you iterate over it. Why is all this index business even required?